pandas/core/groupby/ops.py

```diff
@@ -472,6 +473,29 @@ def _cython_operation(
 
         is_datetimelike = needs_i8_conversion(values.dtype)
         is_numeric = is_numeric_dtype(values.dtype)
+        is_categorical = is_categorical_dtype(values)
+        cat_method_blacklist = (
+            "add",
```

Comment:
why do we need a blacklist at all? you are already only operating on the codes

Reply:
If this method is passed categorical values with a "how" like mean then we don't want to average the codes as if they were ints -- this would happen in the `pandas.tests.groupby.test_function.test_arg_passthru` test without the blacklist, for example. the logic for cases these blacklist cases is already handled in higher in the call stack in `pandas.core.groupby.generic.DataFrameGroupBy._cython_agg_blocks` but maybe we could move some of that logic here for conciseness and/or perf later on

Reply:
i don't think you need this at all, what breaks if you take it out entirely (the blackist)

Reply:
This test will fail without the blacklist: `pandas.tests.groupby.test_function.test_arg_passthru`. Here's a minimal example that will fail:

```python
import pandas as pd

df = pd.DataFrame(
    {"group": [1, 1, 2], "category_string": pd.Series(list("abc")).astype("category")},
    columns=["group", "category_string"],
)

df.groupby("group").mean(numeric_only=False)
```

Nothing else fails so perhaps this test can be updated in lieu of the blacklist, not sure if this is fine from an interface perspective.

Reply:
yeah i think its ok to simply update this test

Reply:
It really seems like we need this blacklist anytime the cython op does something with the actual *values* then we can't pass the codes there. For example something like `mean` or `sum` on a numeric categorical.

Reply:
I see. ok then, let's move this to an attribute on the class, or a function maybe better for easier re-sue

Reply:
Moved to class level in 79f0c72. We could easily make this accessible outside the class with an additional method w/ property decorator